pkg/provisioner/helper_hostpath.go

```diff
@@ -241,71 +252,89 @@ func (pOpts *HelperPodOptions) validateLimits() error {
 //	into pod. Validate that the local pv host path is not created under root.
 func (p *Provisioner) createQuotaPod(ctx context.Context, pOpts *HelperPodOptions) error {
 	var config podConfig
+	var err error
 	config.pOpts, config.podName = pOpts, "quota"
 	// err := pOpts.validate()
-	if err := pOpts.validate(); err != nil {
+	if err = pOpts.validate(); err != nil {
 		return err
 	}
 
 	// Initialize HostPath builder and validate that
 	// volume directory is not directly under root.
 	// Extract the base path and the volume unique path.
-	var vErr error
-	config.parentDir, config.volumeDir, vErr = hostpath.NewBuilder().WithPath(pOpts.path).
+	config.parentDir, config.volumeDir, err = hostpath.NewBuilder().WithPath(pOpts.path).
 		WithCheckf(hostpath.IsNonRoot(), "volume directory {%v} should not be under root directory", pOpts.path).
+		WithCheckf(notReservedVolumeDir(), "volume directory '%s' is a reserved name", config.volumeDir).
 		ExtractSubPath()
-	if vErr != nil {
-		return vErr
+	if err != nil {
```

Comment:
is changing the vErr/lErr necessary?

Reply:
Not necessary, but using single err variable is the idiomatic practice in go.
Example: [Go std](https://github.com/golang/go/blob/release-branch.go1.20/src/net/http/request.go#L555)

Reply:
Yes, we should follow the practice when writing new code. But for the code ported from elsewhere, we should leave it as unchanged as possible.